### PR TITLE
add the cspell in husky

### LINKED ISSUE #562: cspell ci
For some reason cspell doesn't seem to run for any one for the pre-commit hook. 

This bounty is for the cspell to be handled along the husky pre-commit hook.

### PR DESCRIPTION
**Pull request by ghost**
Resolves #562

### CODE REVIEW

**Comment by ghost** [comment]
@pavlovcik can u take a look

**Review by 0x4007 — CHANGES_REQUESTED** [review]
That doesn't seem right because the purpose of `lint-staged` is to run logic on staged files. Why `cspell` the entire codebase when only the changed files need to be `cspell`'d

You should debug `lint-staged`

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
 
-yarn lint-staged --verbose
\ No newline at end of file
+yarn lint-staged
+yarn cspell $"(find . -type f \\( -name '*.sol' -o -name '*.tsx' -o -name '*.ts' \\) ! -path '*/node_modules/*' ! -path '*/lib/*' ! -path '*/types/*')" --no-must-find-files
```
Comment (0x4007) [inline .husky/pre-commit]: Somebody should test this but I think that this runs on every file.

File: package.json [hunk 2]
```diff
@@ -38,22 +38,24 @@
...
     "lint:all": "yarn workspaces foreach run lint",
     "build:ci:netlify": "yarn build:all",
-    "postinstall": "husky install",
-    "lint:cspell:all:watch": "nodemon -e sol,ts,tsx --exec 'yarn lint:cspell:all --fail-fast'",
-    "lint:cspell:all": "cspell $(find . -type f \\( -name '*.sol' -o -name '*.tsx' -o -name '*.ts' \\) ! -path '*/node_modules/*' ! -path '*/lib/*' ! -path '*/types/*')"
+    "postinstall": "husky install"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "yarn lint-staged && yarn cspell $(find . -type f \\( -name '*.sol' -o -name '*.tsx' -o -name '*.ts' \\) ! -path '*/node_modules/*' ! -path '*/lib/*' ! -path '*/types/*') --no-must-find-files",
```
Comment (0x4007) [inline package.json:45]: Pretty sure you're running the logic redundantly here, which makes the commit significantly slower to complete.

**Comment by ghost** [comment]
> That doesn't seem right because the purpose of `lint-staged` is to run logic on staged files. Why `cspell` the entire codebase when only the changed files need to be `cspell`'d
> 
> 
> 
> You should debug `lint-staged`

Like cspell is actually running on just the staged files I checked 🤔 may be try running it locally it should be working

**Comment by 0x4007** [comment]
Unfortunately I need some time to review this because I wrote the bounty five months ago and don't exactly recall what I'm referring to. 

I presume people were pushing changes before without cspell. 

Now that I think about it, it's possible that the husky hook wasn't being modified after old contributors already had the project forked and cloned. 

@molecula451 @rndquu would you guys be able to do a quick test to confirm this hypothesis? 

So to clarify, check if the husky hook gets updated if you already cloned a repository, and yarn installed the dependencies. I believe that it's possible the hook was modified to support cspell but the husky precommit hook wasn't being updated (with cspell) on existing contributor local copies.  

If that's correct then that means the bounty was not necessary in the first place. 

I forgot to spend some days to review and validate all these old bounties.

**Comment by molecula451** [comment]
as @pavlovcik says judging by the input i think this was better for a discussion and feedback, before opening a PR topic on the same issue #562, there was even a closed PR where was recalling the changes "unnecesary" on some reviews https://github.com/ubiquity/ubiquity-dollar/pull/717. That being said I'm taking pavlovcik words and giving a spin-off to this PR to really tackle in what is actually solving and if the issue still exists & persists on the repo after months

**Comment by 0x4007** [comment]
It's possible that it's a non issue though with the new wave of contributors and the new husky `lint-staged` hook already including cspell. It would be great if you can verify that.

**Comment by molecula451** [comment]
This PR is potentially introducing unwanted complexity, while @AnakinSkywalkeer it's doing it's best to meet requirements, the output it's not really clear, as the complexity will mirror the [Spell Check CI](https://github.com/ubiquity/ubiquity-dollar/blob/ed176304f4f89b57ec8975e7bf051261dc6a439f/.github/workflows/spell-check.yml), we are already relying on [Spell Check.yml](https://github.com/ubiquity/ubiquity-dollar/blob/ed176304f4f89b57ec8975e7bf051261dc6a439f/.github/workflows/spell-check.yml) for linting checks, so this husky pre-commit would work out locally, but at the same time adding redundancy to the project. Like right because we already have [spell-check.yml](https://github.com/ubiquity/ubiquity-dollar/blob/ed176304f4f89b57ec8975e7bf051261dc6a439f/.github/workflows/spell-check.yml) and linting check when pushing changes. I have edited the original requirements that pavlovcik wrote as it looks more reasonable from that stand point of view. Beyond that. I would close this PR and issue as not planned, and only re-opened for further discussion when need it. Giving the oportunity to the hunter on a possible future if the issue is re-opened so is we decide to give updates and maintenance to this husky package

**Comment by 0x4007** [comment]
https://chat.openai.com/share/084a9c22-efea-4ad8-93e3-f2c67ca6e66e

> 1. Commit the changes and push to the repository.
> 2. Ask your contributors to pull the latest changes.
> 3. Have them run npx husky install (or yarn husky install) to update the Git hooks on their local repository.

**Comment by molecula451** [comment]
> https://chat.openai.com/share/084a9c22-efea-4ad8-93e3-f2c67ca6e66e
> 
> > 1. Commit the changes and push to the repository.
> > 2. Ask your contributors to pull the latest changes.
> > 3. Have them run npx husky install (or yarn husky install) to update the Git hooks on their local repository.

Looks redudant by the GPT,  this is done by yarning at first sight, because husky is in `package.json` @pavlovcik

**Comment by 0x4007** [comment]
Step 3. says otherwise. 

I think that old contributors need to run `npx husky install` in order to update the commit hooks.

**Comment by molecula451** [comment]
> Step 3. says otherwise.
> 
> I think that old contributors need to run `npx husky install` in order to update the commit hooks.

We'll surely come out with something for this one great pavlovick!

**Comment by rndquu** [comment]
We already use `husky install` in the [postinstall script](https://github.com/ubiquity/ubiquity-dollar/blob/development/package.json#L41) but the postinstall script sometimes is not executed on `yarn install`. There is the long running [issue](https://github.com/yarnpkg/yarn/issues/5476) for this case.

**Pull request closed without merging**